pyproject.toml

```diff
@@ -110,3 +111,5 @@ tokelor = "^0.1"
 astpath = "^0.9.0"
 lxml = "^4.6"
 m2r2 = "^0.2.7"
+
+nbqa = {version = "^0.5.6", python = ">=3.6.1,<4.0"}
```

Comment:
`python` keyword is not required here. `nbqa = "^0.5"` is just fine.

Reply:
Thanks for your review! OK, I've updated `pyproject.toml` - which command should I run such that `poetry.lock` will get updated?

Reply:
`poetry lock` 🙂

Reply:
so easy, thanks :smile: 

If I exclude the Python keyword I get
```
$ poetry lock
Updating dependencies
Resolving dependencies... (13.1s)

  SolverProblemError

  The current project's Python requirement (>=3.6,<4.0) is not compatible with some of the required packages Python requirement:
    - nbqa requires Python >=3.6.1, so it will not be satisfied for Python >=3.6,<3.6.1
    - nbqa requires Python >=3.6.1, so it will not be satisfied for Python >=3.6,<3.6.1
    - nbqa requires Python >=3.6.1, so it will not be satisfied for Python >=3.6,<3.6.1
    - nbqa requires Python >=3.6.1, so it will not be satisfied for Python >=3.6,<3.6.1
    - nbqa requires Python >=3.6.1, so it will not be satisfied for Python >=3.6,<3.6.1
    - nbqa requires Python >=3.6.1, so it will not be satisfied for Python >=3.6,<3.6.1
    - nbqa requires Python >=3.6.1, so it will not be satisfied for Python >=3.6,<3.6.1
  
  Because no versions of nbqa match >0.5,<0.5.1 || >0.5.1,<0.5.2 || >0.5.2,<0.5.3 || >0.5.3,<0.5.4 || >0.5.4,<0.5.5 || >0.5.5,<0.5.6 || >0.5.6,<0.6
   and nbqa (0.5.0) requires Python >=3.6.1, nbqa is forbidden.
  And because nbqa (0.5.1) requires Python >=3.6.1, nbqa is forbidden.
  And because nbqa (0.5.2) requires Python >=3.6.1
   and nbqa (0.5.3) requires Python >=3.6.1, nbqa is forbidden.
  And because nbqa (0.5.4) requires Python >=3.6.1
   and nbqa (0.5.5) requires Python >=3.6.1, nbqa is forbidden.
  So, because nbqa (0.5.6) requires Python >=3.6.1
   and wemake-python-styleguide depends on nbqa (^0.5), version solving failed.

  at venv/lib/python3.8/site-packages/poetry/puzzle/solver.py:241 in _solve
      237│             packages = result.packages
      238│         except OverrideNeeded as e:
      239│             return self.solve_in_compatibility_mode(e.overrides, use_latest=use_latest)
      240│         except SolveFailure as e:
    → 241│             raise SolverProblemError(e)
      242│ 
      243│         results = dict(
      244│             depth_first_search(
      245│                 PackageNode(self._package, packages), aggregate_package_nodes

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties
    
    For nbqa, a possible solution would be to set the `python` property to ">=3.6.1,<4.0"
    For nbqa, a possible solution would be to set the `python` property to ">=3.6.1,<4.0"
    For nbqa, a possible solution would be to set the `python` property to ">=3.6.1,<4.0"
    For nbqa, a possible solution would be to set the `python` property to ">=3.6.1,<4.0"
    For nbqa, a possible solution would be to set the `python` property to ">=3.6.1,<4.0"
    For nbqa, a possible solution would be to set the `python` property to ">=3.6.1,<4.0"
    For nbqa, a possible solution would be to set the `python` property to ">=3.6.1,<4.0"

    https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies,
    https://python-poetry.org/docs/dependency-specification/#using-environment-markers
```
We require 3.6.1+ in nbqa because NamedTuple does not support defaults in 3.6.0

Reply:
Well, this is actually a very "controversial" practice to depend on patch releases. 

Is it possible to fix this from `nbqa`'s side?

Reply:
I share your concern in general, but this particular patch is pretty common, e.g. pre-commit also requires >=3.6.1 https://github.com/pre-commit/pre-commit/blob/master/setup.cfg

Reply:
Ok, let's leave it as is then! Thanks for clarification 👍

Reply:
Hey @sobolevn - ended up fixing this on nbqa's, integration was much simpler now :)